### PR TITLE
Add settings-profile & user-settings-form page objects

### LINKED ISSUE #403: Create Page Object For Settings-Profile
Create page-object for `settings-profile` and update `settings-profile-test.js`

# References
- Progress on: #141

### PR DESCRIPTION
**Pull request by WenInCode**
# Wha't in this PR?
- created a page object for the `settings-profile` & `user-settings-form`.
- updated the `settings-profile` acceptance tests.

# References
- Closes #403 
- Progress on #141

### CODE REVIEW

File: tests/acceptance/settings-profile-test.js [hunk 1]
```diff
@@ -99,12 +100,12 @@ test("it allows editing of users image", (assert) => {
...
+    settingsProfilePage.userSettingsForm.clickSave();
   });
 
   andThen(() => {
-    assert.equal(find('.alert-success').length, 1);
-    assert.equal(find('.alert-success p').text(), "Profile updated successfully");
+    assert.ok(settingsProfilePage.successAlert.isVisible);
+    assert.equal(settingsProfilePage.successAlert.message, "Profile updated successfully");
     let expectedStyle = `url(${droppedImageString})`;
     assert.equal(removeDoubleQuotes(find('.image-drop').css('background-image')), expectedStyle);
```
Comment (WenInCode) [inline tests/acceptance/settings-profile-test.js]: Is there a good way to add this to a page object?

File: tests/acceptance/settings-profile-test.js [hunk 2]
```diff
@@ -99,12 +100,12 @@ test("it allows editing of users image", (assert) => {
 
   andThen(() => {
     fillInFileInput('.image-drop input', { name: fileName, content: droppedImageString });
```
Comment (WenInCode) [inline tests/acceptance/settings-profile-test.js]: Is there a good way to add this to a page object?

Comment (JoshSmith) [inline tests/acceptance/settings-profile-test.js]: Not sure. Same on other comment above. Maybe @begedin could weigh in? Or @green-arrow potentially?

File: tests/acceptance/settings-profile-test.js (hunk 1 above)
Comment (begedin) [inline tests/acceptance/settings-profile-test.js]: For this one, I messed round with it, and where's what I got:

```Javascript
// this is what the entire page object looks like:

import {
  create,
  text,
  visitable,
  findElement
} from 'ember-cli-page-object';
import userSettingsForm from './components/user-settings-form';
import removeDoubleQuotes from '../helpers/remove-double-quotes';


export default create({
  visit: visitable('/settings/profile'),

  successAlert: {
    scope: '.alert-success',
    message: text('p'),
  },

  droppedImage: {
    scope: '.image-drop',
    backgroundImageData() {
      let $el = findElement(this);
      let backgroundImageData = $el.css('background-image');
      return removeDoubleQuotes(backgroundImageData);
    }
  },

  userSettingsForm,
});

// and the assertion:
assert.equal(settingsProfilePage.droppedImage.backgroundImageData(), expectedData);
```

I wanted to use a getter, but it gets evaluated too early, somehow, so I'm not sure what's going on there. Lazy evaluation through a function should be good enough, though.

Comment (begedin) [inline tests/acceptance/settings-profile-test.js]: Don't forget to get rid of unused imports from the test module if you end up going in this direction.

**Comment by JoshSmith** [comment]
@WenInCode what's the status of this one?

**Comment by WenInCode** [comment]
@JoshSmith I didn't realize this had been addressed. I'll take a look after work - thanks for the comments @begedin

**Comment by WenInCode** [comment]
Update this - ready for review again

**Comment by JoshSmith** [comment]
@WenInCode squash when you can and we'll be good to go!

**Pull request merged**